### PR TITLE
Bad configuration tests

### PR DESCRIPTION
**Pull request by rahulanand16nov**
This PR adds test cases for when there is a minor mistake in the config file and checks that the proxy does the intended behavior for these cases.

### CODE REVIEW

File: integration-tests/apisonator.go [hunk 1]
```diff
@@ -376,6 +377,8 @@ func executeHTTPRequest(method string, url string, data *[]byte) (*http.Response
 func (backend *Backend) Push(stateName string, params []interface{}) error {
 	defer func() {
```
Comment (NomadXD) [inline integration-tests/apisonator.go:378]: There is an issue with the use of `defer` here. Defer executes after the surrounding function returns. So here if you do this way, even when the functions in the switch case returns an error , you are adding a new state to the backend states. I fixed this issue in my PR and should be file after you do a rebase.

File: integration-tests/apisonator.go [hunk 2]
```diff
@@ -404,6 +407,8 @@ func (backend *Backend) Pop() error {
 
 		defer func() {
```
Comment (NomadXD) [inline integration-tests/apisonator.go:408]: same issue with the `defer` here.

File: integration-tests/main.go [hunk 3]
```diff
@@ -34,6 +35,7 @@ func GenerateConfig(name string, jsonData []byte) error {
 	if writeErr := ioutil.WriteFile(name, out.Bytes(), 0777); writeErr != nil {
 		fmt.Printf("Error writing temp config file: %v", writeErr)
 	}
+	time.Sleep(100 * time.Millisecond)
```
Comment (NomadXD) [inline integration-tests/main.go:38]: Why we need a timeout here ???

**Review by NomadXD — APPROVED** [review]
LGTM , check comments

File: integration-tests/apisonator.go (hunk 1 above)
Comment (rahulanand16nov) [reply integration-tests/apisonator.go:378]: I was thinking of using named return value to tackle this problem but your solution is also good.

File: integration-tests/apisonator.go (hunk 2 above)
Comment (rahulanand16nov) [reply integration-tests/apisonator.go:408]: I think you have fixed it already :D

File: integration-tests/main.go (hunk 3 above)
Comment (rahulanand16nov) [reply integration-tests/main.go]: I was having issues while testing but forget to remove it before the PR. Good find!

**Pull request merged**